manager/controlapi/ca_rotation.go

```diff
@@ -0,0 +1,216 @@
+	}
+
+	if len(newConfig.SigningCAKey) > 0 && len(newConfig.SigningCACert) == 0 {
+		return grpc.Errorf(codes.InvalidArgument, "if a signing CA key is provided, the signing CA cert must also be provided")
+	}
+
+	// if the desired CA cert and key are not set, then we are happy with the current root CA configuration, unless
+	// the ForceRotate version has been bumped
+	if len(newConfig.SigningCACert) == 0 {
+		if cluster.RootCA.LastForcedRotation < newConfig.ForceRotate {
+			newRootCA, err := ca.CreateRootCA(ca.DefaultRootCN)
+			if err != nil {
+				return grpc.Errorf(codes.Internal, err.Error())
+			}
+			return updateRootRotationObject(ctx, securityConfig, cluster, newRootCA, newConfig.ForceRotate)
+		}
+		return nil
+	}
+
+	// A desired cert and maybe key were provided - we need to make sure the cert and key (if provided) match.
+	var signingCert []byte
+	if newConfig.SigningCAKey != nil {
+		signingCert = newConfig.SigningCACert
+	}
+	newRootCA, err := ca.NewRootCA(newConfig.SigningCACert, signingCert, newConfig.SigningCAKey, ca.DefaultNodeCertExpiration, nil)
+	if err != nil {
+		return grpc.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	if len(newRootCA.Pool.Subjects()) != 1 {
+		return grpc.Errorf(codes.InvalidArgument, "the desired CA certificate cannot contain multiple certificates")
+	}
+
+	// If the desired cert is the same as the current root CA cert then the private signer is the same
+	// whether it's a locally signing CA or an external CA.  The key can be replaced with the desired key (which will
+	// either be the correct key or nil) without having to do any kind of root rotation, so we can abort any outstanding
+	// root rotations and just finish the validation.
+	if bytes.Equal(cluster.RootCA.CACert, newConfig.SigningCACert) {
+		cluster.RootCA.CAKey = newConfig.SigningCAKey
+		cluster.RootCA.RootRotation = nil
+		cluster.RootCA.LastForcedRotation = newConfig.ForceRotate
+		return nil
+	}
+
+	// If the desired cert is the same as the current being-rotated-to CA cert then the private signer is the same
+	// whether it's a locally signing CA or an external CA.  The key can be replaced with the desired key (which will
+	// either be the correct key or nil).
+	if r := cluster.RootCA.RootRotation; r != nil && bytes.Equal(r.CACert, newConfig.SigningCACert) {
+		r.CAKey = newConfig.SigningCAKey
+		cluster.RootCA.LastForcedRotation = newConfig.ForceRotate
+		return nil
+	}
+
+	// if NewRootCA succeeded, parsing will be successful
+	parsedCert, _ := helpers.ParseCertificatePEM(newConfig.SigningCACert)
+
+	// The cert is different than our current cert or currently-rotating-to cert; the new certificate's expiry must be at
+	// least one year away
+	if parsedCert.NotAfter.Before(time.Now().Add(minRootExpiration)) {
+		return grpc.Errorf(codes.InvalidArgument, "CA certificate expires too soon")
+	}
+
+	// If the desired cert has the same public key and subject as current root, then the private signer is the same
```

Comment:
In this case, the cert has changed, it just has the same subject and public key.  One such example is if the cert is renewed.  In this case, leaf certs issued by the previous CA cert will still correctly validate, but the cert digest has changed, and hence join tokens need to be re-generated.